sympy/solvers/solveset.py

```diff
@@ -1754,6 +1768,46 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _compute_lambert_solutions(lhs, rhs, symbol):
+    r"""
+    Computes the equations whose solutions will be given in terms
+    of the list of LambertW function. Returns `None` if it fails doing so.
+
+    Example
+    =======
+
+    >>> from sympy.solvers.solveset import _compute_lambert_solutions
+    >>> from sympy import symbols, exp
+    >>> x = symbols('x')
+    >>> _compute_lambert_solutions(x*exp(x), 1, x)
+    [LambertW(1)]
+
+    """
+    try:
+        poly = lhs.as_poly()
+        gens = _filtered_gens(poly, symbol)
+        return _solve_lambert(lhs - rhs, symbol, gens)
+    except NotImplementedError:
+        pass
+
+
+def _solve_as_lambert(lhs, rhs, symbol, domain):
+    r"""
+    Helper solver to handle equations having Lambert solutions.
+    Initally tries to solve equation directly.
+
+    Notes
+    =====
+    This function needs more improvements.
```

Comment:
This docstring is quite vague.

How about:
```
Find a solution in terms of the Lambert W function if possible.

Solves the equation *lhs* = *rhs* for *symbol* over *domain*.
Returns the solution as a `FiniteSet` or otherwise a `ConditionSet`
if the equation could not be solved.

This function only works if *domain* is a subset of the reals.
```

Reply:
Actually why is this function even needed. Can the logic from here just be moved into `_compute_lambert_solutions`? Perhaps `_compute_lambert_solutions` could be renamed to `_solve_as_lambert`.

Reply:
Yes, it's pretty nice...

Reply:
> Actually why is this function even needed. Can the logic from here just be moved into _compute_lambert_solutions? Perhaps _compute_lambert_solutions could be renamed to _solve_as_lambert.

Yes, that'll be a lot better!